### ISSUE TITLE
add aws_vpc_peering_connections to list all available peering connections

### ISSUE THREAD

**Issue by moshea** [terraform-providers/terraform-provider-aws#6350]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

* Add aws_vpc_peering_connections
list all peering connections using filters, returning ids, cidr_blocks
and peer_cidr_blocks
useful when trying to build route tables between vpcs with multiple
peering connections managed in different state files

### New or Affected Resource(s)

* aws_vpc_peering_connections

### Potential Terraform Configuration

```hcl
# Declare the data source
data "aws_vpc_peering_connections" "pcs" {
  filter {
    name = "requester-vpc-info.vpc-id"
    values = ["${aws_vpc.foo.id}"]
  }
}

# Create a route table
resource "aws_route_table" "rt" {
  vpc_id = "${aws_vpc.foo.id}"
}

# Create a route
resource "aws_route" "r" {
  count                     = "${length(data.aws_vpc_peering_connections.pcs.ids)}"
  route_table_id            = "${aws_route_table.rt.id}"
  destination_cidr_block    = "${element(data.aws_vpc_peering_connections.pcs.peer_cidr_blocks, count.index)}"
  vpc_peering_connection_id = "${element(data.aws_vpc_peering_connections.pcs.ids, count.index)}"
}
```

**Comment by moshea** [comment]
submitted a pull request for this too
https://github.com/terraform-providers/terraform-provider-aws/pull/6351

**Comment by bflad** [comment]
Support for this new data source has been merged and will release with version 3.5.0 of the Terraform AWS Provider, likely on Thursday. Thanks to @moshea and @tmccombs for the implementation. 👍
